Elbow Saver Modules/Tests/ExerciseTimerTests/ExerciseTimerUnitTests.swift

```diff
@@ -0,0 +1,253 @@
+        store.send(.stopButtonTapped) { $0.currentTimerState = .stopped }
+    }
+    
+    func testStopTimerDuringRep() {
+        var initialState = ExerciseTimerState()
+        initialState.currentTimerState = .performingRep
+        let scheduler = DispatchQueue.test
+        let store = TestStore(
+            initialState: initialState,
+            reducer: exerciseTimerReducer,
+            environment: .init(mainQueue: scheduler.eraseToAnyScheduler())
+        )
+        
+        store.send(.stopButtonTapped) { $0.currentTimerState = .stopped }
+    }
+    
+    func testStopTimerBetweenReps() {
+        var initialState = ExerciseTimerState()
+        initialState.currentTimerState = .betweenReps
+        initialState.secondsRemainingForRep = 0
+        let scheduler = DispatchQueue.test
+        let store = TestStore(
+            initialState: initialState,
+            reducer: exerciseTimerReducer,
+            environment: .init(mainQueue: scheduler.eraseToAnyScheduler())
+        )
+        
+        store.send(.stopButtonTapped) { $0.currentTimerState = .stopped }
+    }
+    
+    func testStopTimerBetweenSets() {
+        var initialState = ExerciseTimerState()
+        initialState.currentTimerState = .betweenSets
+        initialState.secondsRemainingForRep = 0
+        initialState.currentRep = ExerciseTimerState.repsPerSet
+        let scheduler = DispatchQueue.test
+        let store = TestStore(
+            initialState: initialState,
+            reducer: exerciseTimerReducer,
+            environment: .init(mainQueue: scheduler.eraseToAnyScheduler())
+        )
+        
+        store.send(.stopButtonTapped) { $0.currentTimerState = .stopped }
+    }
+    
+    func testStopTimerWhenStopped() {
+        var initialState = ExerciseTimerState()
+        initialState.currentTimerState = .stopped
+        let scheduler = DispatchQueue.test
+        let store = TestStore(
+            initialState: initialState,
+            reducer: exerciseTimerReducer,
+            environment: .init(mainQueue: scheduler.eraseToAnyScheduler())
+        )
+        
+        // Pressing Stop when timer is already stopped should NOT mutate timer state.
+        store.send(.stopButtonTapped)
+    }
+    
+    func testStopTimerWhenFinished() {
+        var initialState = ExerciseTimerState()
+        initialState.currentTimerState = .finished
+        initialState.secondsRemainingForRep = 0
+        initialState.currentRep = ExerciseTimerState.repsPerSet
+        initialState.currentSet = initialState.totalNumberOfSets
+        let scheduler = DispatchQueue.test
+        let store = TestStore(
+            initialState: initialState,
+            reducer: exerciseTimerReducer,
+            environment: .init(mainQueue: scheduler.eraseToAnyScheduler())
+        )
+        
+        // Pressing Stop when timer is already finished should NOT mutate timer state.
+        store.send(.stopButtonTapped)
```

Comment:
Can we still pass in `{ $0 = .finished }` to assert that in the closure? Although I guess it's implicit with the `initialState`?

Reply:
We can to be extra explicit, but if we leave the closure out it will still fails if the state changes

Reply:
Might just be clearer what the expected result is, even if it's unnecessary. But totally up to you here